File: CS/WindowsApplication3/DataHelper.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WindowsApplication3
{
    class DataHelper
    {
        ImageList imageList = null;
        public DataHelper(ImageList list)
        {
            imageList = list;
        }

        Image GetImage(int index)
        {
            return imageList.Images[index];
        }

        public DataTable CreateTable()
        {
            int i = 0;
            DataTable table = new DataTable();
            DataRow dataRow;
            table.Columns.Add("CustomerID", typeof(System.Int32));
            table.Columns.Add("FirstName", typeof(System.String));
            table.Columns.Add("LastName", typeof(System.String));
            table.Columns.Add("Image", typeof(System.Drawing.Image));
            table.Columns.Add("Column1", typeof(System.DateTime));
            dataRow = table.NewRow();
            dataRow["CustomerID"] = 0;
            dataRow["FirstName"] = "a/b/c/d";
            dataRow["LastName"] = "0";
            dataRow["Image"] = GetImage(i++);
            dataRow["Column1"] = DateTime.Parse("5/21/2019 12:00:00 AM");
            table.Rows.Add(dataRow);

            dataRow = table.NewRow();
            dataRow["CustomerID"] = 1;
            dataRow["FirstName"] = "a/b/c/d";
            dataRow["LastName"] = "1";
            dataRow["Image"] = GetImage(i++);
            dataRow["Column1"] = DateTime.Parse("5/21/2019 12:00:00 AM");
            table.Rows.Add(dataRow);

            dataRow = table.NewRow();
            dataRow["CustomerID"] = 2;
            dataRow["FirstName"] = "a/b/c/d";
            dataRow["LastName"] = "2";
            dataRow["Image"] = GetImage(i++);
            dataRow["Column1"] = DateTime.Parse("5/21/2019 12:00:00 AM");
            table.Rows.Add(dataRow);

            dataRow = table.NewRow();
            dataRow["CustomerID"] = 3;
            dataRow["FirstName"] = "a/b/c/d";
            dataRow["LastName"] = "3";
            dataRow["Image"] = GetImage(i++);
            dataRow["Column1"] = DateTime.Parse("5/21/2019 12:00:00 AM");
            table.Rows.Add(dataRow);

            dataRow = table.NewRow();
            dataRow["CustomerID"] = 4;
            dataRow["FirstName"] = "a/b/c/d";
            dataRow["LastName"] = "4";
            dataRow["Image"] = GetImage(i++);
            dataRow["Column1"] = DateTime.Parse("5/21/2019 12:00:00 AM");
            table.Rows.Add(dataRow);

            return table;
        }

        public DataTable CreateEmptyTable()
        {
            DataTable table = new DataTable();
            table.Columns.Add("CustomerID", typeof(System.Int32));
            table.Columns.Add("FirstName", typeof(System.String));
            return table;
        }
    }
}

File: VB/WindowsApplication3/DataHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace WindowsApplication3
	Friend Class DataHelper
		Private imageList As ImageList = Nothing
		Public Sub New(ByVal list As ImageList)
			imageList = list
		End Sub

		Private Function GetImage(ByVal index As Integer) As Image
			Return imageList.Images(index)
		End Function

		Public Function CreateTable() As DataTable
			Dim i As Integer = 0
			Dim table As New DataTable()
			Dim dataRow As DataRow
			table.Columns.Add("CustomerID", GetType(System.Int32))
			table.Columns.Add("FirstName", GetType(System.String))
			table.Columns.Add("LastName", GetType(System.String))
			table.Columns.Add("Image", GetType(System.Drawing.Image))
			table.Columns.Add("Column1", GetType(Date))
			dataRow = table.NewRow()
			dataRow("CustomerID") = 0
			dataRow("FirstName") = "a/b/c/d"
			dataRow("LastName") = "0"
'INSTANT VB WARNING: An assignment within expression was extracted from the following statement:
'ORIGINAL LINE: dataRow["Image"] = GetImage(i++);
			dataRow("Image") = GetImage(i)
			i += 1
			dataRow("Column1") = Date.Parse("5/21/2019 12:00:00 AM")
			table.Rows.Add(dataRow)

			dataRow = table.NewRow()
			dataRow("CustomerID") = 1
			dataRow("FirstName") = "a/b/c/d"
			dataRow("LastName") = "1"
'INSTANT VB WARNING: An assignment within expression was extracted from the following statement:
'ORIGINAL LINE: dataRow["Image"] = GetImage(i++);
			dataRow("Image") = GetImage(i)
			i += 1
			dataRow("Column1") = Date.Parse("5/21/2019 12:00:00 AM")
			table.Rows.Add(dataRow)

			dataRow = table.NewRow()
			dataRow("CustomerID") = 2
			dataRow("FirstName") = "a/b/c/d"
			dataRow("LastName") = "2"
'INSTANT VB WARNING: An assignment within expression was extracted from the following statement:
'ORIGINAL LINE: dataRow["Image"] = GetImage(i++);
			dataRow("Image") = GetImage(i)
			i += 1
			dataRow("Column1") = Date.Parse("5/21/2019 12:00:00 AM")
			table.Rows.Add(dataRow)

			dataRow = table.NewRow()
			dataRow("CustomerID") = 3
			dataRow("FirstName") = "a/b/c/d"
			dataRow("LastName") = "3"
'INSTANT VB WARNING: An assignment within expression was extracted from the following statement:
'ORIGINAL LINE: dataRow["Image"] = GetImage(i++);
			dataRow("Image") = GetImage(i)
			i += 1
			dataRow("Column1") = Date.Parse("5/21/2019 12:00:00 AM")
			table.Rows.Add(dataRow)

			dataRow = table.NewRow()
			dataRow("CustomerID") = 4
			dataRow("FirstName") = "a/b/c/d"
			dataRow("LastName") = "4"
'INSTANT VB WARNING: An assignment within expression was extracted from the following statement:
'ORIGINAL LINE: dataRow["Image"] = GetImage(i++);
			dataRow("Image") = GetImage(i)
			i += 1
			dataRow("Column1") = Date.Parse("5/21/2019 12:00:00 AM")
			table.Rows.Add(dataRow)

			Return table
		End Function

		Public Function CreateEmptyTable() As DataTable
			Dim table As New DataTable()
			table.Columns.Add("CustomerID", GetType(System.Int32))
			table.Columns.Add("FirstName", GetType(System.String))
			Return table
		End Function
	End Class
End Namespace
